### PR TITLE
[ RNN ] Add mask for dropout 

### PR DESCRIPTION
**Pull request by jijoongmoon**
This commit includes :
  . implementation of mask for dropout

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/layers/rnn.h [hunk 1]
```diff
@@ -126,6 +127,8 @@ class RNNLayer : public LayerImpl {
    */
   float dropout_rate;
 
+  float epsilon;
```
Comment (kparichay) [inline nntrainer/layers/rnn.h:130]: please add some description for this member variable.

**Review by kparichay — APPROVED** [review]
LGTM

File: nntrainer/layers/rnn.cpp [hunk 2]
```diff
@@ -47,6 +47,11 @@ void RNNLayer::finalize(InitLayerContext &context) {
   output_dim = input_dim;
   output_dim.width(unit);
 
+  if (dropout_rate > epsilon) {
+    wt_idx[RNNParams::dropout_mask] = context.requestTensor(
+      output_dim, "RNN:dropout_mask", true, ITERATION_LIFESPAN);
```
Comment (kparichay) [inline nntrainer/layers/rnn.cpp:52]: `true` -> false`? true is for trainable tensor (which has gradient as well). Do we need gradient for mask tensor itself?

Comment (jijoongmoon) [reply nntrainer/layers/rnn.cpp:52]: is it? I thought it is "assign now" variable. I will fix it. Thanks.

File: nntrainer/layers/rnn.h (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/layers/rnn.h:130]: added. Thanks.

File: nntrainer/layers/rnn.cpp (hunk 2 above)
Comment (kparichay) [reply nntrainer/layers/rnn.cpp:52]: assign now is not part of layer_context interface. It is part of only Var_Grad and used only for internal purpose.

**Review by zhoonit — APPROVED** [review]
LGTM

**Pull request merged**